### PR TITLE
[CAMEL-17841] Add checkpoint batch size & timeout parameters

### PR DESCRIPTION
**Pull request by kevinlacire**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by davsclaus** [comment]
So maybe make a PR that is batch size only as a first PR. And then work on timeout/interval as a 2nd PR that uses a background thread.

**Comment by kevinlacire** [comment]
implemented but not tested yet

**Comment by oscerd** [comment]
@kevinlacire can you please rebase?

**Comment by kevinlacire** [comment]
> @kevinlacire can you please rebase?

I'll take some time to do it tomorrow

**Comment by kevinlacire** [comment]
> 

Do you have an example ? I can take a look to adapt it using Camel's ExecutorServiceStrategy

**Comment by davsclaus** [comment]
Maybe take a look at `LRASagaService` how it creates the thread pool and start/stop it etc.

File: components/camel-azure/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/EventHubsConsumer.java [hunk 1]
```diff
@@ -30,15 +33,25 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static org.apache.camel.component.azure.eventhubs.EventHubsConstants.*;
```
Comment (oscerd) [inline components/camel-azure/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/EventHubsConsumer.java:36]: Please don't import with * operator, use explicit imports.

**Review by oscerd — CHANGES_REQUESTED** [review]
Seems to be good, we should merge in 3.19.0

File: components/camel-azure/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/EventHubsConsumer.java (hunk 1 above)
Comment (kevinlacire) [reply components/camel-azure/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/EventHubsConsumer.java]: done

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
This has been merged

**Pull request closed without merging**